product-service/controllers/products.controller.ts

```diff
@@ -1,23 +1,69 @@
+import { Client } from 'pg';
+
 // models
 import { ProductModel } from '../models/product.model';
 
-// db
-import { MOCK_DB } from '../mocks/db';
-
 export const getAllProducts = async (): Promise<ProductModel[]> => {
-    return new Promise((resolve) => {
-        setTimeout(() => {
-            resolve(MOCK_DB.products);
-        }, 2000);
-    });
+    const dbConnection = new Client();
+
+    try {
+        await dbConnection.connect();
+
+        const { rows: products } = await dbConnection.query(`
+            select id, title, description, price, photo_url, stocks.count from products
+            inner join stocks
+            on products.id = stocks.product_id
+        `);
+
+        return products;
+    } catch (error) {
+        console.log(error);
+    } finally {
+        dbConnection.end();
+    }
+};
+
+export const getProductById = async (productId: string): Promise<ProductModel> => {
+    const dbConnection = new Client();
+
+    try {
+        await dbConnection.connect();
+
+        const { rows: products } = await dbConnection.query(`
+            select id, title, description, price, photo_url, stocks.count from products
+            inner join stocks
+            on products.id = stocks.product_id
+            where id = '${productId}'
+        `);
+
+        return products[0];
+    } catch (error) {
+        console.log(error);
+    } finally {
+        dbConnection.end();
+    }
 };
 
-export const getProductById = async (searchId: string): Promise<ProductModel> => {
-    return new Promise((resolve) => {
-        const product = MOCK_DB.products.find(({ id }) => id === searchId);
+export const insertProduct = async (product: ProductModel): Promise<string> => {
+    const dbConnection = new Client();
+    const prepareQuery = `with first_insert as (
+        insert into products ( title, description, price, photo_url )
+        values ( '${product.title}','${product.description}', '${product.price}', '${product.image}' )
+        returning id
+    )
+    insert into stocks ( product_id , count )
+    values ( ( select id from first_insert ), '${product.count}' )
+    returning product_id`;
+
+    try {
+        await dbConnection.connect();
+
+        const { rows } = await dbConnection.query(prepareQuery);
 
-        setTimeout(() => {
-            resolve(product);
-        }, 2000);
-    });
+        return rows[0].product_id;
+    } catch (error) {
+        console.log(error);
+    } finally {
+        dbConnection.end();
+    }
```

Comment:
this file looks like a service rather than controller

Reply:
Will fix during the next home tasks